test/services/notices/setup/create-alternate-notice.service.test.js

```diff
@@ -0,0 +1,168 @@
+
+// Thing under test
+const CreateAlternateNoticeService = require('../../../../app/services/notices/setup/create-alternate-notice.service.js')
+
+describe('Notices - Setup - Create Alternate Notice service', () => {
+  let fetchFailedReturnsInvitationsResults
+  let fetchReturnsAddressesServiceResults
+  let noticeWithErrors
+  let notifications
+
+  beforeEach(() => {
+    fetchFailedReturnsInvitationsResults = {
+      failedLicenceRefs: ['11/111', '01/124', '01/125'],
+      failedReturnIds: [
+        '18998ffd-feaf-4e24-b998-7e7af026ba14',
+        'c06708f5-195a-43b1-9f2e-d4f72ee7bd76',
+        '0f760d21-0f05-49e7-b226-21ad02dd22b4',
+        '6917b4bb-4b68-480f-ae19-3f525dc7c67b',
+        'e6bc04bc-1899-4b3c-b733-6f4be6aa8e07'
+      ]
+    }
+    fetchReturnsAddressesServiceResults = [
+      _licenceHolderAddress('Anne', '01/111', [generateUUID()]),
+      _licenceHolderAddress('Barry', '01/124', [generateUUID()]),
+      _licenceHolderAddress('Charles', '01/125', [generateUUID()])
+    ]
+
+    noticeWithErrors = {
+      issuer: 'admin-internal@wrls.gov.uk',
+      licences: ['01/123', '01/124', '01/125', '01/126'],
+      metadata: {
+        name: 'Returns: invitation',
+        error: 3,
+        options: { excludeLicences: [] },
+        recipients: 4,
+        returnCycle: {
+          dueDate: '2026-04-28',
+          endDate: '2026-03-31',
+          summer: false,
+          startDate: '2025-04-01',
+          quarterly: false
+        }
+      },
+      overallStatus: 'error',
+      referenceCode: NotificationHelper.generateReferenceCode('RINV'),
+      status: 'completed',
+      subtype: 'returnInvitation',
+      type: 'notification'
+    }
+  })
+
+  afterEach(() => {
+    Sinon.restore()
+  })
+
+  describe('when called with a notice with errors', () => {
+    beforeEach(() => {
+      Sinon.stub(FetchFailedReturnsInvitationsService, 'go').resolves(fetchFailedReturnsInvitationsResults)
+      Sinon.stub(FetchAlternateRecipientsService, 'go').resolves(fetchReturnsAddressesServiceResults)
+    })
+
+    it('returns a list of notifications to be sent and the reference code for the new notice', async () => {
+      const result = await CreateAlternateNoticeService.go(noticeWithErrors)
+
+      notifications = [
+        _notification(result.notice.id, fetchReturnsAddressesServiceResults[0]),
+        _notification(result.notice.id, fetchReturnsAddressesServiceResults[1]),
+        _notification(result.notice.id, fetchReturnsAddressesServiceResults[2])
+      ]
+
+      expect(result.notice.issuer).to.equal(noticeWithErrors.issuer)
+      expect(result.notice.licences).to.equal(['11/111', '01/124', '01/125'])
+      expect(result.notice.metadata.name).to.equal('Returns: invitation')
+      expect(result.notice.metadata.recipients).to.equal(3)
+      expect(result.notice.subtype).to.equal('returnInvitation')
```

Comment:
There are times we check individual properties, but for something like this we'd just compare the result, e.g. `expect(result.notice).to.equal({})`.

You also mentioned checking the instance type. You can do that with `expect(result.notice).to.be.instanceOf(EventModel)`. You'll just need to import `EventModel` as a test helper.